core/src/main/java/com/google/cloud/sql/core/InternalConnectorRegistry.java

```diff
@@ -261,4 +267,55 @@ private Connector createConnector(ConnectionConfig config) {
         refreshTimeoutMs,
         serverProxyPort);
   }
+
+  private ConnectionConfig resolveConfig(ConnectionConfig config) {
+    if (config.getNamedConnection() != null) {
+      ConnectorKey namedConfig = resolveKey(config);
+      if (namedConfig == null) {
+        throw new IllegalArgumentException(
+            "Named configuration " + config.getNamedConnection() + " does not exist.");
+      }
+      return namedConfig.getConfig();
+    }
+    return config;
+  }
+
+  private ConnectorKey resolveKey(ConnectionConfig config) {
```

Comment:
What if it receives only the "String namedConnection" ? There is no need for the entire `ConnectionConfig`.

Reply:
Good idea. Fixed.